template.go

```diff
@@ -160,6 +170,17 @@ func readYAMLRequest(base string, contents []byte, templateArgs map[string]strin
 		opts.ROpts.ThriftFile = thriftFileURL.Path
 	}
 
+	switch strings.ToLower(t.Encoding) {
+	case "thrift":
+		opts.ROpts.Encoding = encoding.Thrift
+	case "json":
+		opts.ROpts.Encoding = encoding.JSON
+	case "protobuf":
+		opts.ROpts.Encoding = encoding.Protobuf
+	case "raw":
+		opts.ROpts.Encoding = encoding.Raw
+	}
+
```

Comment:
Line `169`, there is the following line of code `overrideParam(&opts.ROpts.RequestJSON, string(body))`.  How would it work when the body given in file has another format than JSON? Right now, body could be with format proto, thrift or raw.